### ISSUE TITLE
[iOS YYC] CPP code YYRValue not defined before it is referenced in Nov Beta runtime

### ISSUE THREAD

**Issue by shichen85** [YoYoGames/GameMaker-Bugs#7953]
### Description
When building the sample for YYC, the generated CPP code would fail to build because a YYRValue is referenced before it is defined. This does not happen with the August runtime.

### Steps To Reproduce
1. Build the sample for iOS YYC with runtime 2024.1100.0.661
2. Build the XCode project, see the compile error `Error: use of undeclared identifier 'sWself_validate_bschema_integer_4ABE50B2'`
3. Repeat with runtime 2024.800.0.642, see the XCode project successfully builds

### Which version of GameMaker are you reporting this issue for?
IDE v2024.800.0.618 Runtime v2024.1100.0.661

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22000.0

### Which platform(s) are you seeing the problem on?
iOS, Windows

[e51010ce-6d6d-4a2c-88c2-4810f889288b](https://api.gamemaker.io/api/github/downloads/e51010ce-6d6d-4a2c-88c2-4810f889288b)

**Comment by shichen85** [comment]
Building Windows YYC also fails with the similar error `'sWself_validate_bschema_integer_4ABE50B2': undeclared identifier`

**Comment by shichen85** [comment]
The culprit GML code is written in an unnecessary manner that declares a global function (`_func3`) in the argument position of a function, and we intend to remove it, so we are not expecting a fix for the runtime to accommodate the original GML code. 

```gml
		function _func1(type_string, unique_fields_array, validator_function, hydration_function=undefined) {
			for ( var i = 0; i < array_length(unique_fields_array); i++){
				var _field = unique_fields_array[i];
				validator_function(_field);
			}
		}

		function _func2() {
			_func1("bschema_type_integer", [], 
				function _func3(data, schema, error_collector, json_path) {
					return false;
				}
			);
		}
```

However, because the build error is actually confusing to debug, we would suggest a better compiler/syntax error to be thrown, such as `cannot declare global function in the argument position of another function` and point to the offending GML line.

**Comment by rwkay** [comment]
Fixed in 2024.11
